From c0bbaf1ecab7b858214212f96ac160ea18760130 Mon Sep 17 00:00:00 2001
From: Matt Flax <flatmax@flatmax.org>
Date: Sun, 27 Mar 2022 22:42:47 +1100
Subject: [PATCH] audioinjector.net: stereo and zero, use dev_err_probe, report
 success

Use dev_err_probe to follow best practice, and leave a message in
the kernel log indicate the driver was started successfully.
---
 sound/soc/bcm/audioinjector-pi-soundcard.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/sound/soc/bcm/audioinjector-pi-soundcard.c b/sound/soc/bcm/audioinjector-pi-soundcard.c
index 3f24fa739081..ad337201c558 100644
--- a/sound/soc/bcm/audioinjector-pi-soundcard.c
+++ b/sound/soc/bcm/audioinjector-pi-soundcard.c
@@ -158,9 +158,11 @@ static int audioinjector_pi_soundcard_probe(struct platform_device *pdev)
 			}
 	}
 
-	if ((ret = devm_snd_soc_register_card(&pdev->dev, card))) {
-		dev_err(&pdev->dev, "snd_soc_register_card failed (%d)\n", ret);
-	}
+	if ((ret = devm_snd_soc_register_card(&pdev->dev, card)))
+		return dev_err_probe(&pdev->dev, ret, "%s\n", __func__);
+
+	dev_info(&pdev->dev, "successfully loaded\n");
+
 	return ret;
 }
 
-- 
2.32.0

